File: 프로그래머스/다음큰숫자.py
def solution(n):
    answer = 0
    com = n + 1
    n_com = list(str(format(n,'b')))
    n_count = n_com.count('1')
    while True:
        a = list(str(format(com,'b')))
        if n_count == a.count('1'):
            return com
            break
        com += 1
    

File: 프로그래머스/java/구현/다음큰숫자.java
import java.util.*;


class Solution {

    int cnt = 0;

    public int count(int n){
        if(n == 0){
            return 0;
        }else{
            if (n % 2 == 1){
                this.cnt += 1;
            }
            count(n / 2);
        }
        return this.cnt;
    }

    public int solution(int n) {
        int answer = 0;
        int sol;
        Solution s = new Solution();
        int com = s.count(n);
        s.cnt = 0;

        while(true){
            n += 1;
            if (s.count(n) == com){
                sol = n;
                break;
            }
            s.cnt = 0;
        }
        return sol;
    }
}